src/writer/workflows_blocks/bravesearchworkflow.py

```diff
@@ -0,0 +1,138 @@
+                        "type": "Text",
+                        "description": "The search term or phrase"
+                    },
+                    "location": {
+                        "name": "Location",
+                        "type": "Text",
+                        "description": "Geographic location for search results",
+                        "required": False
+                    },
+                    "device": {
+                        "name": "Device",
+                        "type": "Text",
+                        "options": {
+                            "desktop": "Desktop",
+                            "mobile": "Mobile"
+                        },
+                        "default": "desktop",
+                        "required": False
+                    },
+                    "language": {
+                        "name": "Language",
+                        "type": "Text",
+                        "description": "Language for search results",
+                        "default": "en",
+                        "required": False
+                    },
+                    "num_results": {
+                        "name": "Number of Results",
+                        "type": "Text",
+                        "default": "10",
+                        "required": False
+                    },
+                    "additional_params": {
+                        "name": "Additional Parameters",
+                        "type": "Key-Value",
+                        "default": "{}",
+                        "required": False
+                    }
+                },
+                "outs": {
+                    "success": {
+                        "name": "Success",
+                        "description": "The search was completed successfully.",
+                        "style": "success",
+                    },
+                    "apiError": {
+                        "name": "API Error",
+                        "description": "An error occurred while making the API request.",
+                        "style": "error",
+                    },
+                    "rateLimitError": {
+                        "name": "Rate Limit Error",
+                        "description": "The API rate limit has been exceeded.",
+                        "style": "error",
+                    }
+                },
+            }
+        ))
+
+    def run(self):
+        try:
+            # Get required fields
+            api_key = self._get_field("api_key")
+            query = self._get_field("query")
+            
+            # Get optional fields
+            location = self._get_field("location", True)
+            device = self._get_field("device", True, "desktop")
+            language = self._get_field("language", True, "en")
+            num_results = self._get_field("num_results", True, "10")
+            additional_params = self._get_field("additional_params", True, "{}")
```

Comment:
Some fields are not JSON
```suggestion
            location = self._get_field("location")
            device = self._get_field("device", False, "desktop")
            language = self._get_field("language", False, "en")
            num_results = self._get_field("num_results", False, "10")
            additional_params = self._get_field("additional_params", True, "{}")
```